release_notes.py

```diff
@@ -61,13 +61,33 @@ def issue_link(issue):
         print >>sys.stderr, "Didn't find any issues for the target fix version"
         sys.exit(1)
 
-    unresolved_issues = [issue for issue in issues if issue.fields.resolution is None]
+    # Some resolutions, including a lack of resolution, indicate that the bug hasn't actually been addressed and we shouldn't even be able to create a release until they are fixed
+    UNRESOLVED_RESOLUTIONS = [None,
+                              "Unresolved",
+                              "Duplicate",
+                              "Invalid",
+                              "Not A Problem",
+                              "Not A Bug",
+                              "Won't Fix",
+                              "Incomplete",
+                              "Cannot Reproduce",
+                              "Later",
+                              "Works for Me"
+                              ]
+    unresolved_issues = [issue for issue in issues if issue.fields.resolution in UNRESOLVED_RESOLUTIONS or issue.fields.resolution.name in UNRESOLVED_RESOLUTIONS]
     if unresolved_issues:
-        print >>sys.stderr, "The release is not completed since unresolved issues were found still tagged with this release as the fix version:"
+        print >>sys.stderr, "The release is not completed since unresolved issues or improperly resolved issues were found still tagged with this release as the fix version:"
         for issue in unresolved_issues:
-            print >>sys.stderr, "Unresolved issue: %s %s" % (issue.key, issue_link(issue))
+            print >>sys.stderr, "Unresolved issue: %15s %20s %s" % (issue.key, issue.fields.resolution, issue_link(issue))
+        print >>sys.stderr
+        print >>sys.stderr, "Note that for some resolutions, you should simply remove the fix version as they have not been truly fixed in this release."
         sys.exit(1)
 
+    # Issues with some resolutions should be filtered out because they indicate the bug wasn't valid. Including them in release notes could be misleading
+    IGNORED_RESOLUTIONS = ["Workaround",
+                           "Information Provided"]
```

Comment:
Is it worth treating these differently instead of including them in the UNRESOLVED list? People that look at JIRA release notes for a version would probably think the issue was fixed when it really hasn't.